src/test/java/de/tum/in/www1/artemis/util/HestiaUtilService.java

```diff
@@ -0,0 +1,238 @@
+package de.tum.in.www1.artemis.util;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.connector.BitbucketRequestMockProvider;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.ProgrammingSubmission;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingSubmissionRepository;
+import de.tum.in.www1.artemis.repository.SolutionProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.repository.TemplateProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.service.UrlService;
+import de.tum.in.www1.artemis.service.connectors.GitService;
+
+/**
+ * Utility service specifically used for testing Hestia related functionality.
+ * This currently includes:
+ * - Setting up a template repository
+ * - Setting up a solution repository
+ * - Setting up a test repository
+ *
+ * In the future this service will be extended to make testing of the code hint generation easier.
+ */
+@Service
+public class HestiaUtilService {
```

Comment:
```suggestion
public class HestiaUtilTestService {
```
I think a different name would make obvious in the other places where this service is used that is just a helper service only for the tests.